src/gwt/src/org/rstudio/studio/client/workbench/views/source/model/DocUpdateSentinel.java

```diff
@@ -343,9 +348,19 @@ public void execute()
                if (!oldMarksSpec.equals(newMarksSpec))
                   setProperty("marks", newMarksSpec);
             }
+            
+            // Update cursor selection position after document save
+            String oldRangeSpec = "";
+            if (hasProperty("selection"))
+               oldRangeSpec = getProperty("selection");
+            
+            String newRangeSpec = Range.encode(docDisplay_.getSelectionRange());
+            if (!oldRangeSpec.equals(newRangeSpec))
+               setProperty("selection", newRangeSpec);
```

Comment:
This is going to effectively double the amount of disk I/O for most saves -- `setProperty` calls `modifyDocumentProperties` which eventually calls `source_database::put(...)` on the server, which writes out the whole doc contents to the source database. If we're going to use the save codepath, I think the fold spec would be a better model to follow than the vim marks.